website/docs/guides/upgrade.html.markdown

```diff
@@ -0,0 +1,158 @@
+---
+layout: "nsxt"
+page_title: "VMware NSX Terraform Provider NSX upgrade support"
+description: |-
+  The VMware NSX Terraform Provider NSX upgrade support
+---
+
+# NSX Provider for NSX Upgrade
+
+NSX Terraform Provider offers support for [upgrading NSX components](https://docs.vmware.com/en/VMware-NSX/4.1/upgrade/GUID-E04242D7-EF09-4601-8906-3FA77FBB06BD.html).
+
+The provider support consists of several resources and data sources which can perform a complete upgrade cycle when applied correctly.
+The NSX upgrade process upgrades the manager components of NSX, Edge appliances, and the NSX bits on the compute hosts.
+
+
+## Upgrade process steps
+
+### Preparation for upgrade
+
+Upgrade preparation consists of several actions, and is performed using [nsxt_upgrade_prepare](../resources/upgrade_prepare.html.markdown):
+* Uploading the upgrade and precheck bundles.
+* Acceptance of the user agreement.
+* Reporting of failed pre-checks.
+
+```hcl
+resource "nsxt_upgrade_prepare" "prepare_res" {
+  upgrade_bundle_url    = "http://url-to-upgrade-bundle.mub"
+  precheck_bundle_url   = "http://url-to-precheck-bundle.pub"
+  accept_user_agreement = true
+}
+```
+
+**NOTE:** Acceptance of the license agreement is required to initiate the upgrade process.
+**NOTE:** Precheck bundle is usable only while upgrading a NSX manager of version v4.1.1 and above.
+
+### Acknowledgement of upgrade issues and warnings
+
+The preparation stage may result with various warnings which would halt the upgrade process. These warnings can be 
+suppressed with the `nsxt_upgrade_precheck_acknowledge` resource.
+
+For example:
+```
+There are unacknowledged warnings in precheck: [backupOperationCheck], please address these errors from NSX or using nsxt_upgrade_precheck_acknowledge resource
+```
+
+The resource uses the `precheck_ids` to conclude which prechecks are safe to ignore for this upgrade cycle.
```

Comment:
Can we add a link to NSX docs showing the list of precheck_ids?

Reply:
I couldn't find such document... Maybe @tvigneron could help?

Reply:
As discussed elsewhere, we will figure out a different solution for this.